### DISCUSSION TITLE
[WASM] Can I build wasm on .Net Core 3.1 target ??

### DISCUSSION THREAD

**Discussion by meleeeee** [unoplatform/uno#8888] [Q&A] [answered]
Hello, I'm beginner C# developer. 

I already have Silverlight project but want to migrate it UWP & web service by using Uno platform.
And I'd like to compile it on .Net Core 3.1 target ... But i'm not sure it is possible.

I bumped into error like this.
(sorry for can't attached img... cause company security)

**System.InvalidOperationException: The TargetFramework version must be above 5.0 (found v3.1)**
   Loacation: Uno.Wasm.Bootstrap.ShellTask_vfc89b40db92a14b8f04d4c5c9c72690bdcd7b878.Execute() file D:\a\1\s\src\Uno.Wasm.Bootstrap\ShellTask.cs:줄 210	App2.Wasm	C:\Users\Administrator\.nuget\packages\uno.wasm.bootstrap\3.3.1\build\Uno.Wasm.Bootstrap.targets	175	

So, I'm thinking it is because Uno.Wasm.Bootstrap(3.3.1) version.
And if you can tell me that possibility of complie wasm project on .Net Core 3.1 ... then i'll find the next way..

Thank you.

**Comment by jeromelaban** [comment, marked as answer]
Why do you need to compile on netcoreapp3.1? For wasm, the lowest target supported is net5.0 in all cases, so you won't be able to target lower than this.

**Reply by meleeeee** [reply]
Thank you for reply. I have to apply our own framework (.Net Core 3.1 version) as well for this project.
And I saw your web assembly support .Net Standard 2.0.. 
So I thought I could service netcoreapp3.1 on the web (with webassembly).
Then if I don't migration this project to .net5, there is no way to service this project on the web??

**Reply by jeromelaban** [reply]
Thanks. I'd assume that your framework is a separate project which uses netcoreapp3.1. You can keep that project with `netcoreapp3.1`  without any issues, then reference it from the main project which is set to net5.0.

Indeed we also support project references going down to netstandard1.0 if that is needed.

**Reply by meleeeee** [reply]
Thank you. But If we go with .net5 and reference our .net core 3.1 framework, we will have to revalidate many parts of the framework. How can I get webaseembly support with core 3.1?

**Reply by jeromelaban** [reply]
WebAssembly is not supporting the netcore3.1 runtime (support started with net6), but when running on a new target, revalidating is required in all cases. That being said, netcoreapp3.1, net5 and net6 are compatible in behavior so you should be good to go.

**Comment by meleeeee** [comment]
Hello :) May I ask one more question after the above question? After referencing my framework, the following error message appears

CS0234	'Microsoft.AspNetCore.Mvc.ApplicationParts' 네임스페이스에 'ApplicationPartAttributeAttribute' 형식 또는 네임스페이스 이름이 없습니다. 어셈블리 참조가 있는지 확인하세요.

As you know CS0234 error means 'The type or namespace name 'name' does not exist in the namespace 'namespace' (are you missing an assembly reference?)'
but I didn't add AspNetCore... Once built, it is added suddenly.... Have you ever seen this kind of error ?? If you need more information then i'll add image by mail. 
Thank you

**Reply by jeromelaban** [reply]
There is not enough information here to help you troubleshoot this unfortunately. If you have more specific information, or a repro, we may be able to help.
